Remotion/Data/DomainObjects/Infrastructure/ObjectIDStringSerialization/ObjectIDStringSerializer.cs

```diff
@@ -123,14 +123,30 @@ private ObjectID ParseWithCustomErrorHandler (string objectIDString, Func<string
       var valuePart = objectIDString.Substring(indexOfValuePart, indexOfTypeDelimiter - indexOfValuePart);
       var typePart = objectIDString.Substring(indexOfTypePart);
 
-      Type? type = TypeUtility.GetType(typePart, throwOnError: false);
-      if (type == null)
+#if NET
+      IObjectIDValueParser? valueParser = typePart switch
       {
-        return errorHandler(
-            string.Format("Serialized ObjectID '{0}' is invalid: '{1}' is not the name of a loadable type.", objectIDString, typePart));
+          "System.Guid" => GuidObjectIDValueParser.Instance,
+          "System.Int32" => Int32ObjectIDValueParser.Instance,
+          "System.String" => StringObjectIDValueParser.Instance,
+          _ => null
+      };
+#else
+      IObjectIDValueParser? valueParser = null;
+      var typeParts = typePart.Split(new[] { ',' }, 3);
+      var isFromMscorlib = typeParts.Length < 2 || typeParts[1].Trim() == "mscorlib";
+      if (isFromMscorlib)
+      {
+        valueParser = typeParts[0].Trim() switch
+        {
+            "System.Guid" => GuidObjectIDValueParser.Instance,
+            "System.Int32" => Int32ObjectIDValueParser.Instance,
+            "System.String" => StringObjectIDValueParser.Instance,
+            _ => null
+        };
       }
+#endif
```

Comment:
Question about duplication: this would remove the duplication:
```
#if NETFRAMEWORK
      var typeParts = typePart.Split(new[] { ',' }, 3);
      var isFromMscorlib = typeParts.Length < 2 || typeParts[1].Trim() == "mscorlib";
  var typeName = isFromMscorlib ? typeParts[0].Trim() : null;    
#else
  var typeName =typePart;
#endif

IObjectIDValueParser? valueParser = typeName switch
      {
"System.Guid" => GuidObjectIDValueParser.Instance,
          "System.Int32" => Int32ObjectIDValueParser.Instance,
          "System.String" => StringObjectIDValueParser.Instance,
          _ => null
      };
```

To me, it also reads better because it removes nesting, etc. Thoughts?